### PR TITLE
feat(ux): implement live feedback

### PR DESCRIPTION
**Pull request by GregoryAlbouy**
## Description

Add live feedback from `requester.Requester` while it's running in the CLI.

benchttp-live-feedback

## Changes

- Make `dispatcher.Dispatcher.Run` return context errors instead of discarding them, so we can retrieve that information in `requester.Requester.Run`

### CODE REVIEW

File: dispatcher/dispatcher.go [hunk 1]
```diff
@@ -59,7 +67,7 @@ func (d dispatcher) Do(ctx context.Context, maxIter int, callback func()) error
 	}
 
 	wg.Wait()
-	return nil
```
Comment (c1-ra) [inline dispatcher/dispatcher.go:62]: [question]
Why were we always returning a nil error before? :thinking:

**Review by moreirathomas — APPROVED** [review]
Nice and fancy 🌈 I love it

File: dispatcher/dispatcher.go (hunk 1 above)
Comment (c1-ra) [reply dispatcher/dispatcher.go:62]: Ok, thanks, I was surprised we returned anything in this case, but we're not doing it anymore anyways!

File: requester/requester.go [hunk 2]
```diff
@@ -77,7 +80,13 @@ func (r *Requester) Run() (Report, error) {
 
 	defer cancel()
 
-	if err := dispatcher.New(numWorker).Do(ctx, maxIter, r.record(req, interval)); err != nil {
+	r.start = time.Now()
+	go r.refreshState()
+
+	switch err := dispatcher.New(numWorker).Do(ctx, maxIter, r.record(req, interval)); err {
+	case nil, context.Canceled, context.DeadlineExceeded:
```
Comment (c1-ra) [inline requester/requester.go:87]: Why is `case nil` returning `r.end(err)` and not the same thing as default?

Comment (GregoryAlbouy) [reply requester/requester.go:87]: Here `default` is the case of an _real_ error that prevents the correct execution of `Do`, so it returns an empty returns and the error.
The three values in this `case` L87 are expected and do not prevent the run (`context.DeadlineExceeded` correspond to `globalTimeout` reached for instance), so ihere we end the process the regular way via `r.end`
Then we only use the value of the error to display the status in the CLI: https://github.com/benchttp/runner/blob/71013d338af1dea57966dbd4029b241e28c66bc6/requester/print.go#L73-L86

Is it clearer? 🤔

Comment (c1-ra) [reply requester/requester.go:87]: Ok, thanks for the explanation!
It is clearer!

**Review by c1-ra — APPROVED** [review]
Nice! :+1:

**Review by c1-ra — APPROVED** [review]
:+1: :rainbow:

**Pull request merged**